tests/test_parser_gb.py

```diff
@@ -427,6 +427,21 @@ def test_postal_code(input_data: str, is_match_expected: bool) -> None:
     execute_matching_test(input_data, is_match_expected, data_gb.postal_code)
 
 
+@pytest.mark.parametrize(
+    ("input_data", "expected_postcode"),
+    [
+        ("24 Moorfield Road, IIkley, West Yorkshire LS29 8BL, UK", "LS29 8BL"),
+        ("71 Wilson Avenue Rochester Kent ME1 2SJ", "ME1 2SJ"),
+        ("3 Dyer Street, Cirencester GL7 2PP", "GL7 2PP"),
+    ],
+)
+def test_specific_postcodes(input_data: str, expected_postcode: str) -> None:
```

Comment:
maybe this can use a more descriptive name, like `test_postcodes_with_valid_postcode_as_substring`

Reply:
haha yeah my name sucks https://github.com/Beauhurst/pyap/pull/7/commits/f4306dcfabab091862f66a312f26867f9017f6cb